packages/pglite-react/src/hooks.ts

```diff
@@ -3,13 +3,37 @@ import { Results } from '@electric-sql/pglite'
 import { usePGlite } from './provider'
 import { query as buildQuery } from '@electric-sql/pglite/template'
 
+function paramsEqual(
+  a1: unknown[] | undefined | null,
+  a2: unknown[] | undefined | null,
+) {
+  if (!a1 && !a2) return true
+  if (a1?.length !== a2?.length) return false
+  for (let i = 0; i < a1!.length; i++) {
+    if (!Object.is(a1![i], a2![i])) {
+      return false
+    }
+  }
+  return true
+}
+
 function useLiveQueryImpl<T = { [key: string]: unknown }>(
   query: string,
   params: unknown[] | undefined | null,
   key?: string,
 ): Omit<Results<T>, 'affectedRows'> | undefined {
   const db = usePGlite()
   const [results, setResults] = useState<Results<T>>()
+  const [currentParams, setCurrentParams] = useState(params)
+
+  useEffect(
+    () =>
+      setCurrentParams((currentParams) =>
+        !paramsEqual(currentParams, params) ? params : currentParams,
+      ),
+    [params],
+  )
```

Comment:
@msfstef I think this useEffect will run after the component has rendered. So we get:

1. Update params
2. Render triggered by change, but it uses value of `currentParams` which is old
3. Effect runs after render
4. Render with new params